kola/tests/misc/users.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package misc
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/platform"
+)
+
+func init() {
+	register.Register(&register.Test{
+		Run:         CheckUserShells,
+		ClusterSize: 1,
+		Name:        "coreos.users.shells",
+		UserData:    `#cloud-config`,
+	})
+}
+
+type user struct {
+	username string
+	shell    string
+}
+
+func CheckUserShells(c platform.TestCluster) error {
+	m := c.Machines()[0]
+
+	ValidUsers := []user{
```

Comment:
hm, though this isn't going to work on GCE which adds lots of users. *but* after crawford's update to the GCE oem it will start working if we create instances with the metadata flag to ignore project users/ssh keys.